Library/Homebrew/formula_installer.rb

```diff
@@ -536,18 +569,20 @@ def unbottled_dependencies(deps)
     end
   end
 
+  sig { void }
   def compute_and_install_dependencies
     deps = compute_dependencies
     install_dependencies(deps)
   end
 
+  sig { params(req_map: T::Hash[Formula, T::Array[Requirement]]).void }
   def check_requirements(req_map)
     @requirement_messages = []
     fatals = []
 
     req_map.each_pair do |dependent, reqs|
       reqs.each do |req|
-        next if dependent.latest_version_installed? && req.name == "macos" && req.comparator == "<="
+        next if dependent.latest_version_installed? && req.is_a?(MacOSRequirement) && req.comparator == "<="
```

Comment:
The type guard is necessary bc `comparator` is only defined on `MacOSRequirement`